src/main/scala/com/fasterxml/jackson/module/scala/ser/EitherSerializer.scala

```diff
@@ -1,95 +1,174 @@
-      case (Some(v: AnyRef), Some(vs), _) => vs.serialize(v, jgen, provider)
-      case (Some(v), _, Some(ts)) => provider.findValueSerializer(v.getClass, beanProperty.orNull).serializeWithType(v, jgen, provider, ts)
-      case (Some(v), _, _) => provider.findValueSerializer(v.getClass, beanProperty.orNull).serialize(v, jgen, provider)
-      case (None, _, _) => provider.defaultSerializeNull(jgen)
+  override def createContextual(prov: SerializerProvider, prop: BeanProperty): JsonSerializer[_] = {
+    val propOpt = Option(prop)
+
+    val newLeft = createContextualDetails(prov, prop, left)
+    val newRight = createContextualDetails(prov, prop, right)
+
+    // Also: may want to have more refined exclusion based on referenced value
+    val newIncl = propOpt match {
+      case None => contentInclusion
+      case Some(p) =>
+        val pinc = p.findPropertyInclusion(prov.getConfig, classOf[Option[AnyRef]])
+        val incl = pinc.getContentInclusion
+        if (incl != JsonInclude.Include.USE_DEFAULTS) {
+          Some(incl)
+        } else contentInclusion
     }
+    withResolved(propOpt, newLeft, newRight, newIncl)
+  }
+
+  override def serialize(value: Either[AnyRef, AnyRef], jgen: JsonGenerator, provider: SerializerProvider): Unit = {
+    serialize(value, jgen, provider, None)
   }
 
-  def createContextual(prov: SerializerProvider, property: BeanProperty): JsonSerializer[_] = {
-    // Based on the version in OptionSerializer
-    def serializerFromAnnotation(property: BeanProperty, serializerProvider: SerializerProvider) = {
-      Option(property).flatMap(p => Option(p.getMember)).flatMap { m =>
-        Option(serializerProvider.getAnnotationIntrospector.findContentSerializer(m))
-          .map(serDef => serializerProvider.serializerInstance(m, serDef))
-      }
+  def serialize(value: Either[AnyRef, AnyRef], jgen: JsonGenerator, provider: SerializerProvider, vts: Option[TypeSerializer]): Unit = {
+    val (field, content, details) = value match {
+      case Left(c) => ("l", c, left)
+      case Right(c) => ("r", c, right)
     }
-    def hasContentTypeAnnotation(provider: SerializerProvider, property: BeanProperty) =
-      Option(property).exists { p =>
-        Option(provider.getAnnotationIntrospector.refineSerializationType(provider.getConfig, p.getMember, p.getType)).isDefined
-      }
-
-    def tryContentSerializer(serializerProvider: SerializerProvider, property: BeanProperty, currentSer: Option[JsonSerializer[_]]) = {
-      val ser = Option(findConvertingContentSerializer(serializerProvider, property, currentSer.orNull))
-      (ser, elementType) match {
-        case (None, Some(et)) if hasContentTypeAnnotation(serializerProvider, property) =>
-          Option(serializerProvider.findValueSerializer(et, property))
-        case (Some(sr), _) => Option(serializerProvider.handleSecondaryContextualization(sr, property))
-        case _ => ser
-      }
+    jgen.writeStartObject()
+    jgen.writeFieldName(field)
```

Comment:
Works for now; going forward the next thing to support would be inclusion (JsonInclude.Include) checks, which would apply here.